### PR TITLE
Generate fake release versions for RPM builds better

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Instad of using a Bash-based approach to parsing and bumping the version
that Origin produces in `${OS_GIT_VERSION}`, a simple Go-based approach
can give us a more stable and readable utility.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@smarterclayton not having this broke GCE

**Comment by soltysh** [comment]
That and missing 1.6 tag, which is needed to have newer rpms installed. Although this has some bugs, I'm currently addressing.

File: tools/versionbump/versionbump.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
+package main
```
Comment (smarterclayton) [inline tools/versionbump/versionbump.go]: Can't our tito tagger do this?

Comment (stevekuznetsov) [inline tools/versionbump/versionbump.go]: We have to tell the tagger what version to tag. `tito` has no utilities for bumping versions AFAICT

/cc @dgoodwin

**Comment by stevekuznetsov** [comment]
[test]
[testextended][extended:core(build)]

**Comment by stevekuznetsov** [comment]
[test_install]

**Comment by stevekuznetsov** [comment]
@bparees could you take a gander at the failures in the build extended tests here please:
 - https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended/1/
 - https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended/2/

**Comment by bparees** [comment]
@stevekuznetsov all 4 extended test failures failed on:

```
20:28:44 STEP: Waiting for a default service account to be provisioned in namespace
```

these are all part of the api version compatibility tests that @csrwng added, so he might be able to tell what you what makes them special.  (they try to run older versions of the api, so they are probably doing some weird stuff)

File: tools/versionbump/versionbump.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	currentSemver.Build = sanitizedMetadata
+
+	// for RPM, we want to have a post-release patch level of 0, which we can fake with Pre
+	currentSemver.Pre = []semver.PRVersion{
+		semver.PRVersion{
+			VersionStr: "0",
+		},
+	}
+
+	fmt.Fprintf(os.Stderr, "%s\n", currentSemver.String())
```
Comment (soltysh) [inline tools/versionbump/versionbump.go]: You need to print to stdout, not stderr.

File: tools/versionbump/versionbump.go (hunk 1 above)
Comment (dgoodwin) [inline tools/versionbump/versionbump.go]: Tito's version tagger doesn't need to be told what version, by default it just bumps whatever is in the spec at that time.

**Comment by stevekuznetsov** [comment]
Closing in favor of the output of the Trello card tracking this feature

**Pull request closed without merging**